include/audio/dmic.h

```diff
@@ -0,0 +1,301 @@
+	u8_t	max_pdm_clk_dc;
+
+	/* parameters unique to each PDM controller */
+
+	/* Bit mask to optionally invert PDM clock */
+	u8_t	pdm_clk_pol;
+	/* Bit mask to optionally invert mic data */
+	u8_t	pdm_data_pol;
+	/* Collection of clock skew values for each PDM port */
+	u32_t	pdm_clk_skew;
+};
+
+/**
+ * Configuration of the PCM streams to be output by the PDM hardware
+ */
+struct pcm_stream_cfg {
+	/*
+	 * if either rate or width is set to 0 for a stream,
+	 * the stream would be disabled
+	 */
+
+	/* PCM sample rate of stream */
+	u32_t			pcm_rate;
+	/* PCM sample width of stream */
+	u8_t			pcm_width;
+	u8_t			reserved;
+	/* PCM sample block size per transfer */
+	u16_t			block_size;
+	/* SLAB for DMIC driver to allocate buffers for stream */
+	struct k_mem_slab	*mem_slab;
+};
+
+/**
+ * Mapping/ordering of the PDM channels to logical PCM output channel
+ */
+struct pdm_chan_cfg {
+	/*
+	 * mapping of PDM controller and mic channel to logical channel
+	 * since each controller can have 2 audio channels (stereo),
+	 * there can be total of 8x2=16 channels.
+	 * The actual number of channels shall be described in
+	 * pcm_stream_cfg.num_chan.
+	 * if 2 streams are enabled, the channel order will be the same for
+	 * both streams
+	 * Each channel is described as a 4 bit number, the least significant
+	 * bit indicates LEFT/RIGHT selection of the PDM controller.
+	 * The most significant 3 bits indicate the PDM controller number.
+	 *	bits 0-3 are for channel 0, bit 0 indicates LEFT or RIGHT
+	 *	bits 4-7 are for channel 1, bit 4 indicates LEFT or RIGHT
+	 *	and so on.
+	 * CONSTRAINT: The LEFT and RIGHT channels of EACH PDM controller needs
+	 * to be adjacent to each other.
+	 */
+	/* Requested channel map */
+	u32_t	req_chan_map_lo;	/* Channels 0 to 7 */
+	u32_t	req_chan_map_hi;	/* Channels 8 to 15 */
+	/* Actual channel map that the driver could configure */
+	u32_t	act_chan_map_lo;	/* Channels 0 to 7 */
+	u32_t	act_chan_map_hi;	/* Channels 8 to 15 */
+	/* requested number of channels */
+	u8_t	req_num_chan;
+	/* Actual number of channels that the driver could configure */
+	u8_t	act_num_chan;
+	/* requested number of streams for each channel */
+	u8_t	req_num_streams;
+	/* Actual number of streams that the driver could configure */
+	u8_t	act_num_streams;
+};
+
+/**
+ * Input configuration structure for the DMIC configuration API
+ */
+struct dmic_cfg {
+	struct pdm_io_cfg io;
+	/*
+	 * Array of pcm_stream_cfg for application to provide
+	 * configuration for each stream
+	 */
+	struct pcm_stream_cfg *streams;
+	struct pdm_chan_cfg channel;
+};
+
+/**
+ * Function pointers for the DMIC driver operations
+ */
+struct _dmic_ops {
+	int (*configure)(struct device *dev, struct dmic_cfg *config);
+	int (*trigger)(struct device *dev, enum dmic_trigger cmd);
+	int (*read)(struct device *dev, u8_t stream, void **buffer,
+			size_t *size, s32_t timeout);
+};
+
+/**
+ * @brief Build the channel map to populate struct pdm_chan_cfg
```

Comment:
Doxygen brief explicitly used while previous elements had brief done using javadoc autobrief style.
I don't mind any but its better to be consistent.

Reply:
Thanks for the input. I'll make this consistent.